### PR TITLE
Thread safe fitting in EcalUncalibRecHitRecAnalFitAlgo

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Switch to ROOT API's that are thread safe as this class is used in a stream module.
This was found by the IB static analysis information.

#### PR validation:

Code compiles

### CODE REVIEW

File: RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h [hunk 1]
```diff
@@ -15,10 +15,14 @@
 #include <string>
 
 #include "TROOT.h"
-#include "TMinuit.h"
+//#include "TMinuit.h"
```
Comment (mandrenguyen) [inline RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h:18]: I propose to clean the commented include here

File: RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h [hunk 2]
```diff
@@ -119,13 +128,13 @@ class EcalUncalibRecHitRecAnalFitAlgo : public EcalUncalibRecHitRecAbsAlgo<C> {
     pedestal.SetParameter(0, frame[0]);
     pedestal.SetParName(0, "Pedestal");
 
-    graph.Fit(&pulseShape, "QRM");
+    int result = graph.Fit(&pulseShape, "QRMN SERIAL");
     //TF1 *pulseShape2=graph.GetFunction("pulseShape");
```
Comment (mandrenguyen) [inline RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h:132]: ... and maybe take the occasion to clean up this one as well

File: RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h [hunk 3]
```diff
@@ -119,13 +128,13 @@ class EcalUncalibRecHitRecAnalFitAlgo : public EcalUncalibRecHitRecAbsAlgo<C> {
...
 
-    if (std::string(gMinuit->fCstatu.Data()) == std::string("CONVERGED ")) {
+    if (0 == result) {
       double amplitude_value = pulseShape.GetParameter(0);
 
-      graph.Fit(&pedestal, "QRL");
+      graph.Fit(&pedestal, "QRLN SERIAL");
       //TF1 *pedestal2=graph.GetFunction("pedestal");
```
Comment (mandrenguyen) [inline RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h:138]: ... and this one
Thanks!

**Review by mandrenguyen — CHANGES_REQUESTED** [review]

**Comment by Dr15Jones** [comment]
please test

File: RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h (hunk 3 above)
Comment (Dr15Jones) [reply RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h]: done

File: RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h (hunk 2 above)
Comment (Dr15Jones) [reply RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h]: done

File: RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h (hunk 1 above)
Comment (Dr15Jones) [reply RecoLocalCalo/EcalRecAlgos/interface/EcalUncalibRecHitRecAnalFitAlgo.h]: done

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
please test

**Comment by mandrenguyen** [comment]
+1 
technical PR

**Comment by perrotta** [comment]
+1

**Pull request merged**